src/_pytest/mark/structures.py

```diff
@@ -344,25 +345,22 @@ def with_args(self, *args: object, **kwargs: object) -> "MarkDecorator":
         mark = Mark(self.name, args, kwargs, _ispytest=True)
         return MarkDecorator(self.mark.combined_with(mark), _ispytest=True)
 
-    # Type ignored because the overloads overlap with an incompatible
-    # return type. Not much we can do about that. Thankfully mypy picks
-    # the first match so it works out even if we break the rules.
     @overload
-    def __call__(self, arg: Markable) -> Markable:  # type: ignore[misc]
```

Comment:
I wonder how come this error doesn't happen anymore. The overloads still overlap with an incompatible return type.

Reply:
I believe it changed when I changed the below to use `Any` instead of `object`, but I can double check that and see if I may have bumped the mypy version I was testing with.